From 6e1fd1ce3d3507e559df41cbbb3d4ea94982d408 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Fri, 21 Jun 2024 13:32:46 +0300
Subject: [PATCH] [SW-189578] gaudi3: tpm: fix wrong gaudi3_pcr.bin gen

The boot fit hash file used to generate the PCR
binary was moved during a folder restructuring.
This relocation was not reflected in the PCR
binary generation logic, causing incorrect PCR
binary generation.

This patch updates the path in the PCR binary
generation logic to ensure the correct file is
used.

Change-Id: Ifddb5921d19848ab4ed2e853d46ea769110c08ec
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/board/habanalabs/gaudi3/gen-pcr-info.sh b/board/habanalabs/gaudi3/gen-pcr-info.sh
index 8ab0a29..901e217 100644
--- a/board/habanalabs/gaudi3/gen-pcr-info.sh
+++ b/board/habanalabs/gaudi3/gen-pcr-info.sh
@@ -2,6 +2,7 @@
 
 HASH_IMGS_LABEL="img-hash"
 ARC_IMAGES_DIR=${BR2_EXTERNAL}/output/${HL_DEV}_arc/images
+IMAGE_ITBS_DIR=${BR2_EXTERNAL}/output/${HL_DEV}/images/itbs
 # A PCR list contains at least 8 entries, and we use 2,
 # thus the rest are padded with '0s' as such:
 # (TOTAL_DIGESTS - USED_DIGESTS) x sizeof(TPM2B_DIGEST) x (chars per byte)  # 6 x 66 x 2 = 792
@@ -9,5 +10,5 @@
 
 IMG_PATHS+=" ${ARC_IMAGES_DIR}/`ls -1 ${ARC_IMAGES_DIR} | grep ${HASH_IMGS_LABEL} |\
 			 grep images-pointers`"
-IMG_PATHS+=" ${ARC_IMAGES_DIR}/`ls -1 ${ARC_IMAGES_DIR}| grep ${HASH_IMGS_LABEL} | \
+IMG_PATHS+=" ${IMAGE_ITBS_DIR}/`ls -1 ${IMAGE_ITBS_DIR}| grep ${HASH_IMGS_LABEL} | \
 			 grep ${HL_DEV}-boot-fit`"
